pkg/acs/calacs/lib/mkspt.c

```diff
@@ -191,6 +214,27 @@ int mkNewSpt (char *in_list, char *mtype, char *output) {
         freeHdr (&header);
     }
 
+    /* If necessary, re-open the primary header to update the NEXTEND keyword */
+    if ((extnum != numSPT) || (numSPT != nimgs)) {
+
+        im = openUpdateImage (out_spt, "", 0, &header);
```

Comment:
The first use of "im" is closed via closeImage(im) before the call to openUpdateImage().

Reply:
Sure, but you had to check if that was really so.

Reply:
Yes, I had to check!  Several options, but in the end I chose to create a new IODescPtr which I deemed to be the most recognizable solution for most folks.  Having done this (and after our conversation), scope power rocks.

Reply:
This is cool as is but it's still better to get into the habit of scoping. Appart from controlling memory lifetimes, it conveys a more concrete message of intent to future developers - the message being that the scoped variable has no need to persist beyond its scoped block. This can also help constrain namespace.

Reply:
...in addition, it also prevents the next developer from stumbling at the same point & same manner by trying to reuse it. They just won't be able to and therefore won't try doing so, the scope guides them.